map-guessr/index.ts

```diff
@@ -0,0 +1,813 @@
+        text: "パラメータ",
+      },
+    },
+    {
+      type: "section",
+      text: {
+        type: "mrkdwn",
+        text: `*範囲の1辺の長さ*\n出題される問題では正方形(${img_size}px×${img_size}px)の衛星画像が表示されます。その正方形の1辺に対応する地球上の距離を指定できます。単位はkmです。正でない数や10000を超える数が指定された場合と指定がない場合は1000kmに設定されます。`,
+      },
+    },
+    {
+      type: "section",
+      text: {
+        type: "mrkdwn",
+        text: "*国・地域*\n出題される問題の中心点の位置を国・地域単位で限定できます。国・地域の先頭に-を付けるとその国・地域を除外します。指定がない場合、世界全体が設定されます。-付きの国・地域のみが指定された場合、世界全体からそれらの国・地域を除いた範囲に設定されます。",
+      },
+    },
+    {
+      type: "header",
+      text: {
+        type: "plain_text",
+        text: "注意点・その他",
+      },
+    },
+    {
+      type: "section",
+      text: {
+        type: "mrkdwn",
+        text: "画像の中心点は陸上に限定されています。Google Mapのズームの限界値より小さい範囲を指定した場合は、Google Mapのズームの限界値の範囲となります。\n解答では度分秒形式はサポートしていません。35度7分30秒であれば35.125のように小数で答えてください。また、北緯、南緯、東経、西経はそれぞれN, S, E, Wで表し、数字の前につけてください。小文字でもよいです。N, S, E, W, n, s, e, wのいずれもついていない数字は緯度、経度の順で解釈され、それぞれ正の数が北緯あるいは東経、負の数が南緯あるいは西経と解釈します。\n*例*\nN65.4E34.8 → 北緯65.4度東経34.8度\n24.6 -98.4 → 北緯24.6度西経98.4度\nw 178.5,32.9 → 北緯32.9度西経178.5度（wが付いている数字が経度と解釈され、残った数字が緯度と解釈されます。アルファベットの後にスペースがあってもよいです。）\n134.9 s17.6 → 南緯17.6度東経134.9度（sが付いている数字が緯度と解釈され、残った数字が経度と解釈されます。）",
+      },
+    },
+    {
+      type: "section",
+      text: {
+        type: "mrkdwn",
+        text: "`座標当て countries`あるいは `座標あて countries`と入力すると、対応している国・地域の一覧が見られます。",
+      },
+    },
+  ],
+  channel: CHANNEL,
+};
+
+function countriesExpand(
+  countriesOriginal: string[],
+  aliases: Record<string, string[]>
+) {
+  const countries = countriesOriginal.concat();
+  let countriesLength = 0;
+
+  while (countries.length !== countriesLength) {
+    countriesLength = countries.length;
+    for (let i = 0; i < countriesLength; i++) {
+      if (aliases.hasOwnProperty(countries[i])) {
+        countries.push(...aliases[countries[i]]);
+        countries[i] = "-";
+      }
+      if (
+        countries[i].startsWith("-") &&
+        aliases.hasOwnProperty(countries[i].slice(1))
+      ) {
+        countries.push(
+          ...Array.from(aliases[countries[i].slice(1)], (x: string) => "-" + x)
+        );
+        countries[i] = "-";
+      }
+    }
+  }
+  return countries;
+}
+
+function countriesListGen(aliases: Record<string, string[]>): string[] {
+  const arr = [];
+
+  const aliasesString = [];
+
+  let aliasesStringLength = 0;
+
+  for (const [key, value] of Object.entries(aliases)) {
+    const addString = `*${key}* → ${value.sort().join(" ")}`;
+    if (aliasesStringLength + addString.length <= 2000) {
+      aliasesString.push(addString);
+      aliasesStringLength += addString.length;
+    } else {
+      arr.push(aliasesString.join("\n"));
+      aliasesString.splice(0);
+      aliasesStringLength = 0;
+    }
+  }
+  arr.push(aliasesString.join("\n"));
+  return arr;
+}
+
+const postOptions = JSON.parse(
+  JSON.stringify({
+    username: "coord-quiz",
+    icon_emoji: ":globe_with_meridians:",
+  })
+);
```

Comment:
[MAY] objectのクローンをしているけど、かなり不要そう

```suggestion
const postOptions = {
  username: "coord-quiz",
  icon_emoji: ":globe_with_meridians:",
};
```